week-3/slideshow/slideshow.js

```diff
@@ -1,8 +1,81 @@
 const images = [
-    "./assets/cute-cat-a.png",
-    "./assets/cute-cat-b.jpg",
-    "./assets/cute-cat-c.jpg",
+  "./assets/cute-cat-a.png",
+  "./assets/cute-cat-b.jpg",
+  "./assets/cute-cat-c.jpg",
+  "./assets/cute-cat-d.jpg",
+  "./assets/cute-cat-e.webp",
+  "./assets/cute-cat-f.jpg",
+  "./assets/cute-cat-g.jpg",
+  "./assets/cute-cat-h.webp",
+  "./assets/cute-cat-i.jpg",
+  "./assets/cute-cat-j.jpg",
+  "./assets/cute-cat-k.jpg",
+  "./assets/cute-cat-l.jpg",
+  "./assets/cute-cat-m.webp",
 ];
 
+// Write your code here
+const backward = document.getElementById("backward-btn");
+const forward = document.getElementById("forward-btn");
+const imageCarousel = document.getElementById("carousel-img");
+const autoBackward = document.getElementById("auto-back-btn");
+const autoForward = document.getElementById("auto-forward-btn");
+const stop = document.getElementById("stop-btn");
+const inputTime = document.getElementById("inputTime");
+imageCarousel.style.height = "30rem";
+imageCarousel.style.width = "50rem";
+let autoPlay; //This is created to allow the use of set interval...
 
-// Write your code here
\ No newline at end of file
+let index = 0;
+
+function displayImg(page) {
+  imageCarousel.src = images[page];
+}
+
+backward.addEventListener("click", () => {
+  if (index === 0) {
+    index = images.length - 1;
+    displayImg(index);
+  } else {
+    index--;
+    displayImg(index);
+  }
+});
+
+autoBackward.addEventListener("click", () => {
+  autoPlay = setInterval(() => {
+    if (index === 0) {
+      index = images.length - 1;
+      displayImg(index);
+    } else {
+      index--;
+      displayImg(index);
+    }
+  }, inputTime.value * 1000);
+});
+
+stop.addEventListener("click", () => {
+  clearInterval(autoPlay);
+});
+
+forward.addEventListener("click", () => {
+  if (index === images.length - 1) {
+    index = 0;
+    displayImg(index);
+  } else {
+    index++;
+    displayImg(index);
+  }
+});
+
+autoForward.addEventListener("click", () => {
+  autoPlay = setInterval(() => {
+    if (index === images.length - 1) {
+      index = 0;
+      displayImg(index);
+    } else {
+      index++;
+      displayImg(index);
+    }
+  }, inputTime.value * 1000);
+});
```

Comment:
I ran into a bug while testing this code. Steps to reproduce the bug - 
1. Launch index.html.
2. Add any numeric value for input time.
3. Press AutoForward button.
4. Press AutoBackward button.
5. Press the stop button.

What happens when you replicate the above mentioned steps? How would you fix this bug or the unexpected behaviour that happens?